azure-active-directory-server/src/main/java/org/jetbrains/teamcity/aad/AADAuthenticationScheme.java

```diff
@@ -121,7 +122,12 @@ public HttpAuthenticationResult processAuthenticationRequest(@NotNull HttpServle
     final String email = token.getClaim(EMAIL_CLAIM);
 
     final ServerPrincipal principal = myPrincipalFactory.getServerPrincipal(name, oid, email, schemeProperties);
-
+   
+    if(principal == null)
```

Comment:
we don't need this check since teamcity core will determine if it should delegate user creation to another auth scheme in this case or create new user